game/definitions.rpy

```diff
@@ -803,6 +854,84 @@ init python in utils:
         """
         return datetime.datetime.now().hour
 
+    def get_is_weekday():
+        """
+        Gets whether the current day is a weekday (Monday : Friday).
+
+        OUT:
+            True if weekday, otherwise False
+        """
+        return datetime.datetime.now().weekday() < 5
+
+    def get_current_time_block():
+        """
+        Returns a type describing the current time of day as a segment.
+        """
+        if get_current_hour() in range(3, 5):
+            return TIME_BLOCK_EARLY_MORNING
+
+        elif get_current_hour() in range(5, 9):
+            return TIME_BLOCK_MID_MORNING
+        
+        elif get_current_hour() in range(9, 12):
+            return TIME_BLOCK_LATE_MORNING
+
+        elif get_current_hour() in range(12, 18):
+            return TIME_BLOCK_AFTERNOON
+
+        elif get_current_hour() in range(18, 22):
+            return TIME_BLOCK_EVENING
+
+        else:
+            return TIME_BLOCK_NIGHT
+
+    def get_holiday_for_date(input_date=None):
```

Comment:
Personally prefer to avoid abbreviations where possible myself - a bit more verbose but less chance of being misconstrued, as well as more consistent with the other date-related items

Reply:
in that case, `input_datetime` as it specifies what the type should be I'd say